app/controllers/concerns/course/statistics/reference_times_concern.rb

```diff
@@ -17,28 +17,28 @@ def personal_end_at_hash(assessment_id_array, course_id)
       ),
 
       personal_times AS (
-        SELECT cu.id AS course_user_id, pt.end_at, pt.assessment_id
+        SELECT cu.id AS course_user_id, pt.start_at, pt.end_at, pt.fixed, pt.assessment_id
         FROM (
           SELECT course_users.id
           FROM course_users
           WHERE course_users.course_id = #{course_id}
         ) cu
         LEFT JOIN (
-          SELECT course_user_id, end_at, assessment_id
+          SELECT course_user_id, start_at, end_at, fixed, assessment_id
           FROM course_user_personal_end_at
         ) pt
         ON cu.id = pt.course_user_id
       ),
```

Comment:
Why need nested subquery?
Can we do:
```sql
personal_times AS (
  SELECT cu.id AS course_user_id, pt.start_at, pt.end_at, pt.fixed, pt.assessment_id
  FROM course_users cu
  LEFT JOIN course_user_personal_end_at pt
  ON cu.id = pt.course_user_id
  WHERE cu.course_id = #{course_id}
)```

Reply:
Same for the other queries

Reply:
when making this nested subquery, my consideration is to make the code neater to read (having the code just like how u wrote it looks very condensed and not really that nice to read).

However, I think this is just about preference and I have adjusted the code to be like what you suggested above